internal/dlsym/dlsym.go

```diff
@@ -0,0 +1,37 @@
+package dlsym
+
+// #cgo LDFLAGS: -ldl
```

Comment:
you can write `#cgo !no_libdl LDFLAGS: -ldl`, but I just tested on debian 12, MacOS and alpine (musl), and all (at least in recent versions) work _both_ with`-ldl` or without. So that doesn't seem to be an issue anyway.

Reply:
Good to know. If there is a need for a `no_dl` build-tag we can add that here too.

Reply:
So you want to keep it? Are there cases, where we need it?

Reply:
Yes, I want to keep it. I removed it and lots of CI jobs failed because of it (but it worked on my laptop?).